extra_data/tests/test_reader_mockdata.py

```diff
@@ -489,6 +489,17 @@ def test_select(mock_fxe_raw_run):
     assert 'FXE_DET_LPD1M-1/DET/0CH0:xtdf' in sel2.instrument_sources
 
 
+def test_select_require_all(mock_sa3_control_data):
+    run = H5File(mock_sa3_control_data).select('*/BEAMVIEW2:daqOutput')
+    subrun = run.select('*/BEAMVIEW2:daqOutput', require_all=True)
```

Comment:
Can we include another source, with data for all trains, so the test checks that it's correctly getting the intersection of train IDs (as opposed to e.g. the union across selected sources)? Even the CONTROL data of the same camera should do.

Reply:
You're right, that's actually the most relevant use case. I've refactored the test to try several different selections including `*`.